wolfcrypt/src/asn.c

```diff
@@ -8282,7 +8412,32 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
             }
         }
     }
-
+#if defined(WOLFSSL_RENESAS_TSIP)
+    /* prepare for TSIP TLS cert verification API use */
+    if (cert->keyOID == RSAk) {
+        /* to call TSIP API, it needs keys position info in bytes */
+        RsaPublicKeyDecodeRawIndex(cert->publicKey, (word32*)&idx,
+                                   cert->pubKeySize,
+                                   &cert->sigCtx.pubkey_n_start,
+                                   &cert->sigCtx.pubkey_n_len,
+                                   &cert->sigCtx.pubkey_e_start,
+                                   &cert->sigCtx.pubkey_e_len);
+        cert->sigCtx.certBegin = cert->certBegin;
+    }
+    /* check if we can use TSIP for cert verification */
+    /* if the ca is verified as tsip root ca.         */
+    /* TSIP can only handle 2048 bits(256 byte) key.  */
+    if (cert->ca && tsip_checkCA(cert->ca->cm_idx) != 0 &&
+        cert->sigCtx.pubkey_n_len == 256) {
+        
+        /* assign memory to encrypted tsip Rsa key index */
+        if (!cert->tsip_encRsaKeyIdx)
+            cert->tsip_encRsaKeyIdx = (byte*)XMALLOC(560,
+                             cert->heap, DYNAMIC_TYPE_RSA);
+        if (cert->tsip_encRsaKeyIdx == NULL)
+                return MEMORY_E;
+    }
```

Comment:
Should there be an else block here in the case where tsip_checkCA() or key length fails?

Reply:
If tsip_checkCA() or length are failure, we would want not to use TSIP certificate verification, but use software certificate verification. Therefore, we would want to set cert->tsip_encRsaKeyIdx that is the last argument of ConfirmSignature to NULL obviously at else-section.

Reply:
Can you add a debug message in the case we fall back to Software?  This will at least let users know from the log that TSIP API was not able to be used.

Reply:
Added a debug message.